src/main/java/com/augustnagro/utf8/LookupTables.java

```diff
@@ -6,108 +6,128 @@
-    byte2HighTable[10] = TOO_LONG | OVERLONG_2 | TWO_CONTS | SURROGATE | TOO_LARGE;
-    byte2HighTable[11] = TOO_LONG | OVERLONG_2 | TWO_CONTS | SURROGATE | TOO_LARGE;
-
-    byte2HighTable[12] = TOO_SHORT;
-    byte2HighTable[13] = TOO_SHORT;
-    byte2HighTable[14] = TOO_SHORT;
-    byte2HighTable[15] = TOO_SHORT;
-    byte2HighLookup = ByteVector.fromArray(species, byte2HighTable, 0);
-
-
-    byte[] incompleteTable = new byte[species.length()];
-    Arrays.fill(incompleteTable, 0, incompleteTable.length - 3, (byte) 0);
-    incompleteTable[incompleteTable.length - 3] = (byte) 0b11110000;
-    incompleteTable[incompleteTable.length - 2] = (byte) 0b11100000;
-    incompleteTable[incompleteTable.length - 1] = (byte) 0b11000000;
-    isIncompleteAnd = ByteVector.fromArray(species, incompleteTable, 0);
-
-    byte[] incompleteEq = new byte[species.length()];
-    Arrays.fill(incompleteEq, 0, incompleteEq.length - 3, Byte.MIN_VALUE);
-    incompleteEq[incompleteEq.length - 3] = (byte) 0b11110000;
-    incompleteEq[incompleteEq.length - 2] = (byte) 0b11100000;
-    incompleteEq[incompleteEq.length - 1] = (byte) 0b11000000;
-    isIncompleteEq = ByteVector.fromArray(species, incompleteEq, 0);
-  }
+    /* Lookup table definitions */
+    private static final byte TOO_SHORT = 1 << 0;
+    private static final byte TOO_LONG = 1 << 1;
+    private static final byte OVERLONG_3 = 1 << 2;
+    private static final byte SURROGATE = 1 << 4;
+    private static final byte OVERLONG_2 = 1 << 5;
+    private static final byte TWO_CONTS = (byte) (1 << 7);
+    private static final byte TOO_LARGE = 1 << 3;
+    private static final byte TOO_LARGE_1000 = 1 << 6;
+    private static final byte OVERLONG_4 = 1 << 6;
+    private static final byte CARRY = TOO_SHORT | TOO_LONG | TWO_CONTS;
+
+    private static final ByteVector byte1HighLookup_128 = generateByte1HighLookup(ByteVector.SPECIES_128);
+    private static final ByteVector byte1LowLookup_128 = generateByte1LowLookup(ByteVector.SPECIES_128);
+    private static final ByteVector byte2HighLookup_128 = generateByte2HighLookup(ByteVector.SPECIES_128);
+    private static final ByteVector isIncompleteAnd_128 = generateIsIncompleteAnd(ByteVector.SPECIES_128);
+    private static final ByteVector isIncompleteEq_128 = generateIsIncompleteEq(ByteVector.SPECIES_128);
+
+    private static final ByteVector byte1HighLookup_256 = generateByte1HighLookup(ByteVector.SPECIES_256);
+    private static final ByteVector byte1LowLookup_256 = generateByte1LowLookup(ByteVector.SPECIES_256);
+    private static final ByteVector byte2HighLookup_256 = generateByte2HighLookup(ByteVector.SPECIES_256);
+    private static final ByteVector isIncompleteAnd_256 = generateIsIncompleteAnd(ByteVector.SPECIES_256);
+    private static final ByteVector isIncompleteEq_256 = generateIsIncompleteEq(ByteVector.SPECIES_256);
+
+    public static final ByteVector byte1HighLookup =
```

Comment:
Do you think the user would want to specify the Vector size and do they gain any benefit by it?  It seems kinda pointless at this point to have this instead of just using `ByteVector.SPECIES_PREFERRED`.  However, if any other point in implementation if a vector size dependent step needs to be performed, it can be worthwhile applying this technique at there.

Reply:
> if any other point in implementation if a vector size dependent step needs to be performed, it can be worthwhile applying this technique at there.

Sounds good. It really is knee-capping that the JVM doesn't consider `final` to be `final`. I was hoping it would be fixed by jdk 17 but not yet..

> Do you think the user would want to specify the Vector size and do they gain any benefit by it? It seems kinda pointless at this point to have this instead of just using ByteVector.SPECIES_PREFERRED

Yes, because SPECIES_PREFERRED is defined as the largest supported vector size, which is not necessarily the fastest. The bigger the chunk size the more likely we fail to take the ASCII fast-path. But for now it doesn't really matter.